traffic_ops/testing/api/v14/user_test.go

```diff
@@ -61,11 +63,10 @@ func CreateTestUsers(t *testing.T) {
 	for _, user := range testData.Users {
 
 		resp, _, err := TOSession.CreateUser(&user)
-		log.Debugln("Response: ", resp.Alerts)
-
 		if err != nil {
-			t.Errorf("could not CREATE user: %v\n", err)
+			t.Fatalf("could not CREATE user: %v\n", err)
```

Comment:
@rob05c Nearly two months ago @dangogh changed Fatalf to Errorf across all tests because Fatalf tests from cleaning up (2928/commits/f32c5ec73db68ef9b24caa7ad6a0fb175051eaf0). It seems that other tests would have the same panic issue if their create method failed. I want to make sure we have a consistent understanding of how tests should terminate and why.

Reply:
Full link: https://github.com/apache/trafficcontrol/pull/2928/commits/f32c5ec73db68ef9b24caa7ad6a0fb175051eaf0

Reply:
+1 -- if the Fatalf occurs before the test gets cleaned up,  the following test may fail.   Really should only use Fatalf if we can't recover from the situation,  and that should be made very clear.

Reply:
>Really should only use Fatalf if we can't recover from the situation

That's exactly what happens here, that's why I changed it. If an error is returned, the following line
```
log.Debugln("Response: ", resp.Alerts)
```

will panic. If an error is returned, the test can't keep going, which is exactly what `Fatalf` is for. Because of the panic, the test won't get cleaned up anyway, and subsequent tests will fail anyway. Using `Fatal` instead of `Error` avoids the panic stacktrace, and makes it easier to see what went wrong to fix it.

We really need to fix the API Test framework to clean up tests even if they panic or `t.Fatalf`. I don't think it'll be hard, I think it's as simple as changing all our tests from e.g.
```
func TestCacheGroups(t *testing.T) {
	CreateTestTypes(t)
	CreateTestCacheGroups(t)
	GetTestCacheGroups(t)
	CheckCacheGroupsAuthentication(t)
	UpdateTestCacheGroups(t)
	DeleteTestCacheGroups(t)
	DeleteTestTypes(t)
}
```
To
```
func TestCacheGroups(t *testing.T) {
	defer DeleteTestTypes(t)
	CreateTestTypes(t)
	defer DeleteTestCacheGroups(t)
	CreateTestCacheGroups(t)
	GetTestCacheGroups(t)
	CheckCacheGroupsAuthentication(t)
	UpdateTestCacheGroups(t)
}
```
(similar to https://github.com/apache/trafficcontrol/blob/d083557e1a59cd4620b7652fd2821bf5d18d1ad3/traffic_ops/testing/api/v14/origins_test.go#L24 but the `defer Delete`s need to come before their `Create`s).

But that should probably be its own PR.

Reply:
Actually, thinking about this some more, I think we should use `Fatalf` anywhere the test can't continue, irrespective of `defer` cleanups attempts. 

There are always going to be cases where "delete" can't clean up appropriately, after a test fails at some point. Test errors are just like compile errors: you should only ever consider the first failure. Because any one failure can cause a chain of other unrelated failures, or worse, false successes. Any one test failure is a failure of the entire test suite. 

The only right answer is to only consider the first failure, and cleanup is irrelevant, because all subsequent tests, pass or fail, are invalid.

This means all build pipelines (and people) should be using `go test -failfast`.